nomos-services/network/src/backends/libp2p/swarm/mod.rs

```diff
@@ -220,13 +218,16 @@ impl SwarmHandler {
     // TODO: Consider a common retry module for all use cases
     fn retry_connect(&mut self, connection_id: ConnectionId) {
         if let Some(mut dial) = self.pending_dials.remove(&connection_id) {
-            dial.retry_count += 1;
+            dial.retry_count = dial
+                .retry_count
+                .checked_add(1)
+                .expect("retry count overflow");
```

Comment:
Nitpick: there is a check for retry count > MAX_RETRY and retry_count is usize. Do we really need checked_add?

Reply:
Actually, we only check for `>` relationships, so in the case that 1. `MAX_RETRY` is `usize::MAX` and we reach that value, we would pass the test since `usize::MAX <= usize::MAX` and when we go increment again before storing we overflow. So in the case `MAX_RETRY` is actually `usize::MAX`, but doing a simple addition we would never get out of the recursion.

Reply:
Maybe I am missing something but in the checked_add case we would panic in this edge case, no?

Maybe we could maybe reorder them like:
```
if dial.retry_count >= MAX_RETRY {  // Check first
    return;
}
dial.retry_count += 1;  
```

Reply:
Yes, it would panic instead of being stuck in an infinite loop. But we can change the panic to a return instead. Maybe we should do that.

Reply:
```
if dial.retry_count >= MAX_RETRY {  // Check first
    return;
}
```

If you do this and then increment after, it should cover the case with MAX_RETRY is usize:MAX and return when it performs usize:MAX retries

Reply:
There will soon (not so soon) be a lint coming that will get rid of all potential overflows, since they are indeed quite sneaky in release builds and especially for small numeric types like `u8`s. So I think I'll leave the checked operation.